### PR TITLE
Bug nil status fix. Config changed

### LINKED ISSUE #5: Status 'nil' when an error is handled
Status 'nil' when an error is handled and this rescue send an status.

### PR DESCRIPTION
**Pull request by mcolombo87**
## Summary

* Fix #5. 
* Change way to config Wprof (with support for the old way)
* Fix subscribers init (Now they start with an initializer).
* New test.

## Notes
* Response to issue #5

### CODE REVIEW

File: lib/generators/wprof_generator.rb [hunk 1]
```diff
@@ -11,32 +11,34 @@ def init_wprof
...
+      ## DEFAULT: nil
+      # config.external_headers = { headers: {'User-Agent' => 'Httparty'}}
+      ## Define your Custom Methods here! (Must be an Array)
+      # config.custom_methods = [ 'my_great_method' ]
+      ## Define your own path for save Files when Reporter Type is FILE
+      ## By default Wor-Prof use Rails.root
+      # config.file_path = '/home/mcolombo/examplefolder'
     end
+    ## Initiate Wprof
+    WprofSubscriptors.subscriptors
```
Comment (mtejedorwolox) [inline lib/generators/wprof_generator.rb]: creo que prefiero que se llame dentro del modulo Configuration, que el usuario no pueda borrar esa linea y sacar los subscriptors

Comment (mcolombo87) [reply lib/generators/wprof_generator.rb]: Done.

**Review by mtejedorwolox — APPROVED** [review]

File: lib/wprof/conf_wprof.rb [hunk 2]
```diff
@@ -1,5 +1,16 @@
 module WProf
-  class Config
+  module Configuration
```
Comment (holywyvern) [inline lib/wprof/conf_wprof.rb]: Not a motive to reject, but we tend to write when multiple static methods exists as a:

```rb
class << self
end
```

block instead.

If you do so, you can use things like private

File: lib/wprof/conf_wprof.rb [hunk 3]
```diff
@@ -9,17 +20,26 @@ class Config
...
-        end
-        DEFAULTS_CONFIGS[param]
+    module_function
+
+    DEFAULTS_CONFIGS.each do |key, value|
+      define_method key do
+        instance_variable_get("@#{key}") || instance_variable_set("@#{key}", value)
+      end
+
+      define_method "#{key}=" do |v|
```
Comment (holywyvern) [inline lib/wprof/conf_wprof.rb]: this could be replaced by `attr_writer key`

**Review by holywyvern — APPROVED** [review]

**Pull request merged**